backend/molgenis-emx2-sql/src/main/java/org/molgenis/emx2/sql/SqlQuery.java

```diff
@@ -670,11 +670,29 @@ private Field<Object> jsonGroupBySelect(
         Column c = isValidColumn(table, field.getColumn());
         List<Field> subselectFields = new ArrayList<>();
         // need pkey to allow for joining of the subqueries
-        subselectFields.addAll(
-            table.getPrimaryKeyFields().stream()
-                .map(f -> f.as(name("pkey_" + f.getName())))
-                .collect(Collectors.toList()));
-        selectFields.add(c.getJooqField());
+        table
```

Comment:
These 20 lines look very complex for just looping over the keys and their references and collecting some names. But that may just be my lack of java experience.